codegen/src/main/resources/software/amazon/awssdk/codegen/rules/AwsEndpointProviderUtils.java.resource

```diff
@@ -119,6 +119,10 @@ public final class AwsEndpointProviderUtils {
             return builder.build();
         } else if (value instanceof Value.Str) {
             String errorMsg = value.expectString();
+            if (errorMsg.contains("Invalid ARN") && errorMsg.contains(":s3:::")) {
```

Comment:
Based on the [error message format](https://github.com/aws/aws-sdk-java-v2/blob/master/services/s3/src/main/resources/codegen-resources/endpoint-rule-set.json#L7589), if it contains the ":s3:::" pattern, then it can only have been a simple bucket ARN being passed in.

Reply:
Here you are completely over writing the errorMsg.
User should be able to lnow the "Unrecognized format: arn:aws:s3:::bucketName (type: bucketName)" from the error message.

Reply:
Good point, updating to keep original message as well